File: Excel/553 Challenge.py
import pandas as pd
import numpy as np

path = "553 ASCII Abacus.xlsx"
test = pd.read_excel(path, header=None, usecols="B:T", nrows=12, skiprows=1).values

M = np.full((12, 19), np.nan, dtype=object)

for i in range(12):
    for j in range(19):
        if i in [0, 4, 11]:
            M[i, j] = "---"
        elif i in [1, 5]:
            M[i, j] = "|"
        else:
            M[i, j] = "O"

for i in range(1, 11):
    M[i, 0] = "|"
    M[i, 18] = "|"

print(np.all(M == test)) # True

File: Excel/500-599/553/553 Challenge.R
library(tidyverse)
library(readxl)

path = "Excel/553 ASCII Abacus.xlsx"
test  = read_excel(path, range = "B2:T13", col_names = FALSE) %>% as.matrix()

M = matrix(NA_character_, nrow = 12, ncol = 19)

for (i in 1:12) {
  for (j in 1:19) {
    if (i %in% c(1, 5, 12)) {
      M[i, j] = "---"
    } else if (i %in% c(2, 6)) {
      M[i, j] = "|"
    } else {
      M[i, j] = "O"
    }
  }
}

for (i in 2:11) {
  M[i, 1] = "|"
  M[i, 19] = "|"
}

all(M == test) %>% print()
as.data.frame(M)
